src/include/btree.i

```diff
@@ -579,7 +587,7 @@ __wt_page_modify_clear(WT_SESSION_IMPL *session, WT_PAGE *page)
 	 * Allow the call to be made on clean pages.
 	 */
 	if (__wt_page_is_modified(page)) {
-		page->modify->write_gen = 0;
+		page->modify->page_state = WT_PAGE_CLEAN;
```

Comment:
Do we need a barrier here to make sure it is visible?

Reply:
No - the timing of when other threads see this change of state isn't important

Reply:
Maybe a comment about why that is here would be good since every other use of the `page_state` field either uses atomic ops or a barrier and comments about it.

Reply:
I think you, Keith Alex and I should get on a call to talk about the need for barriers in this code early next week

Reply:
52269c9